apps/laboratory/src/utils/BitcoinUtil.ts

```diff
@@ -0,0 +1,141 @@
+import type { BitcoinConnector } from '@reown/appkit-adapter-bitcoin'
+import type { CaipNetwork, CaipNetworkId } from '@reown/appkit'
+import * as networks from '@reown/appkit/networks'
+import * as bitcoin from 'bitcoinjs-lib'
+
+export const BitcoinUtil = {
+  createSignPSBTParams(params: BitcoinUtil.CreateSignPSBTParams): BitcoinConnector.SignPSBTParams {
+    const network = this.getBitcoinNetwork(params.network.caipNetworkId)
+
+    const psbt = new bitcoin.Psbt({ network })
+    const payment = bitcoin.payments.p2wpkh({ address: params.senderAddress, network })
+
+    if (!payment.output) {
+      throw new Error('Invalid payment output')
+    }
+
+    const change = this.calculateChange(params.utxos, params.amount, params.feeRate)
+
+    if (change < 0) {
+      throw new Error('Insufficient funds')
+    } else if (change > 0) {
+      psbt.addOutput({
+        address: params.senderAddress,
+        value: BigInt(change)
+      })
+    }
+
+    for (const utxo of params.utxos) {
+      psbt.addInput({
+        hash: utxo.txid,
+        index: utxo.vout,
+        witnessUtxo: {
+          script: payment.output,
+          value: BigInt(utxo.value)
+        }

```

Comment:
is this always a witness utxo? doesn't it depend on the key type?

Reply:
What do you mean by key type?

Reply:
https://www.hiro.so/blog/understanding-the-differences-between-bitcoin-address-formats-when-developing-your-app

Reply:
maybe we should check the address before using `p2wpkh`?